### DISCUSSION TITLE
[Feature] Auto create shared albums to face match users  / add new face matches to existing albums

### DISCUSSION THREAD

**Discussion by vulcanjedi** [immich-app/immich#3306] [Feature Request]
### Auto create shared albums to face match users / add new face matches to existing albums

Like Google Photos...all new photos are scanned for face match and those w/ predefined tagged individuals able to share / added to shared album (that way family member can auto get new photos of grandkids, new baby...etc)
Having to manually vet each Face Match and then add (new) photos to each shared album is laborious.  
Even w/ users with full access (currently they can only see the timeline and not search against it) so for family sharing I think this is a critical feature.
Push/App Notifications of new photos in shared album would be nice (if not already?)
Saves a lot of duplicate storage sending photos / vids in family group text chats.

Im the family curator / custodian of photos/vids for entire extended family.  I have to troll entire catalog.
For events like graduations, anniversaries other projects nice to click on a person to find their media (assuming face matching is decent).
Huge time saver.
Instantly sharing newborn, kids events to extended family like aunts/uncles/cousins/grandparents.....etc. or just making it easier to troll when convenient for whomever.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by marcinjakubowski** [comment]
My use case exactly, want to share my kids' photos with their grandmother automatically.

**Comment by vulcanjedi** [comment]
How is this feature not voted up more. With the current sharing mechanisms, I still think the sharing lacks much utility at all to just simply see an unsearchable shared unified timeline of a 'partners' library. 
And manually shared folders are cumbersomely laborious currently with having to manually add, not integrated w/ face match, and search doesn't seem to search unsortable shared folders or shared/partnered timelines.
Vote this up

**Comment by irrigate-hanks** [comment]
Voted up. Also the exact use case for me. I believe it is the last feature that would allow me to completely cut ties with Google Photos once and for all.

**Comment by livilivi** [comment]
vote up.  And also want an option to allow auto-shared photos to appear in Timeline.

**Comment by iamernie** [comment]
Voted Up.  This is a much-needed feature when you don't want to share all of your photos with your partner but want to share photos of specific family members/people.  Hopefully, this gets road-mapped.

**Comment by zyxzxyx** [comment]
Is there a plan for this feature? This exactly what is preventing me from cutting off G Photos.

**Comment by mikegrogan** [comment]
Came here to try to see if there was any discussion about this. Seems like all the pieces are almost in place. This would be a great addition. Only reason I can't completely switch off G Photo as well just yet.

**Comment by romainrbr** [comment]
I am also looking forward to such a feature being integrated into Immich !

In the meantime, I've created a small tool that allows you to do that using Immich's API : https://pypi.org/project/immich-face-to-album/

**Reply by nickstrand** [reply]
This looked like it might work and be nice originally, but it looks like that API endpoint limits to 1000 results and I don't see any pagination options, mind you I don't know a ton about apis.  Either way it'll work for 1k and then you'll never get any more even if you cron it, just a heads up

**Reply by pmdTorrents** [reply]
https://pastebin.com/Vi5hA10r

i think this works. apparently you can use time buckets with the person_id assets.

**Reply by projek01** [reply]
it works!!! thank you very much for your help

**Comment by bo0tzz** [comment]
Duplicate of  #1673

**Reply by vulcanjedi** [reply]
respectfully not sure I see the duplicity

**Reply by ajnart** [reply]
I guess the #1673 is kind of a superset of this feature request, what it asks for is much more complex. Keeping this discussion open would be great tho as it already has 60+ people interested in it. They could help clarify what it is about.

**Comment by vulcanjedi** [comment]
Right this request has 600% more votes and specificity the ‘original’ thread is far more abstract and has less votes
________________________________

**Comment by Malnes** [comment]
This is the only feature that keeps me and my family from comitting to immich.

**Comment by projek01** [comment]
I agree with everyone here. I actually bought a bunch of hardware to run immich thinking that this feature was already a thing. I hope the developers can implement it soon. Thanks for all of the hard work.

**Comment by will9512** [comment]
Current solution (based on [ixtrader](https://github.com/ixtrader) code :


https://pastebin.com/79C7iX1T

fixed :)

dont forget to change the entries:
server_url = "https://immich.yurserver.xy/"
email = "[test@test.xy](mailto:test@test.xy)"
password = "yuorpassword"
person_id = "1dde757d-00c5-4aab-b610-934a1105f1f0"
album_id = "c5cb853e-ae21-48c3-aeea-b8efe72da9c0"




_this script will do it for you using the immich api

https://pastebin.com/YmbKwXeF

you just need to fill in:
    server_url =     (https://localhost:port)
    email =     (immich login)
    password =    (immich pass)
    user_id =    (https://localhost:port/people/**this number***)
    album_id =   (https://localhost:port/albums/**this number***)
    
and set it to run every x amount of time using either windows task manager or crontab_

**Reply by jimski427** [reply]
would we need multiple scripts if we wanted to do multiple faces to albums?

**Reply by pmdTorrents** [reply]
as it is now, yes

**Reply by projek01** [reply]
thank you! this works for me and is the last thing I needed to cancel my oogle subscription!

**Reply by projek01** [reply]
Is there a way to include videos in this script? I just noticed that videos of said person don't get added to the album. Thanks!

*edit - I figured it out. I did a "ctrl-f" and replaced 'IMAGE' with 'VIDEO' and saved as another script that does the video assets.

**Reply by pmdTorrents** [reply]
Is there a way to include videos in this script? I just noticed that videos of said person don't get added to the album. Thanks!

    *edit - I figured it out. I did a "ctrl-f" and replaced 'IMAGE' with 'VIDEO' and saved as another script that does the video assets.

nice :), I just added that on my end this week too, i think it used to work for videos too but either way here is updated script
https://pastebin.com/QJzBNAci

**Reply by projek01** [reply]
Darn, I spoke too soon. This script is limited to 1000 assets. Did you figure out any workarounds?

**Reply by pmdTorrents** [reply]
https://pastebin.com/Vi5hA10r

i think this works. apparently you can use time buckets with the person_id assets.

**Comment by ajb3932** [comment]
Do you mind if I ask why this is a closed thread? It doesn't look like this feature has been implimented natively which is the request?

**Comment by EuleMitKeule** [comment]
I'd love this feature, please reopen it!

**Reply by alextran1502** [reply]
I think it is duplicated with another feature request i.e smart album so it is closed

**Reply by EuleMitKeule** [reply]
In the meanwhile, this works great: https://github.com/romainrbr/immich-face-to-album

**Comment by ixtrader** [comment]
The api has changed.

the new solution python solution is here:
https://pastebin.com/vHtRb9Lk

dont forget to change the entries:
   server_url =     "https://immich.yurserver.xy"
    email =     "test@test.xy"
    password =    "yuorpassword"
    person_id =   "1dde757d-00c5-4aab-b610-934a1105f1f0"
    album_id =   "c5cb853e-ae21-48c3-aeea-b8efe72da9c0"

**Reply by CrispinPZA** [reply]
Sadly, this no longer works. 


```
Failed to fetch time buckets. Status code: 404, Response text: {"message":"Cannot GET /api/asset/time-buckets?userId=1d8aaaa4-e847-45ae-86d2-2d0bbbbbb1340&size=MONTH","error":"Not Found","statusCode":404,"correlationId":"j0rm2ep2v"}
Total unique assets: 0
```

**Reply by will9512** [reply]
https://pastebin.com/79C7iX1T

fixed :)

dont forget to change the entries:
server_url = "https://immich.yurserver.xy/"
email = "[test@test.xy](mailto:test@test.xy)"
password = "yuorpassword"
person_id = "1dde757d-00c5-4aab-b610-934a1105f1f0"
album_id = "c5cb853e-ae21-48c3-aeea-b8efe72da9c0"

**Reply by CrispinPZA** [reply]
Awesome - thanks for that! Works well.

**Reply by randyesperben** [reply]
> The api has changed.
> 
> the new solution python solution is here: https://pastebin.com/vHtRb9Lk
> 
> dont forget to change the entries: server_url = "https://immich.yurserver.xy" email = "[test@test.xy](mailto:test@test.xy)" password = "yuorpassword" person_id = "1dde757d-00c5-4aab-b610-934a1105f1f0" album_id = "c5cb853e-ae21-48c3-aeea-b8efe72da9c0"

Hi! does this code work with the latest version of Immich and its API changes?

**Comment by harssh** [comment]
I am in favor of this feature. This would make everyone in my family move from Google Photos to Immich.

**Comment by Alihkhawaher** [comment]
Me too, it is one of the most must have features. In google photos I auto share my family photos. It a simple and the most fun feature a photo app can have. Please add this feature. I too also thought this app had it, I already migrated everything to immich.

Thanks

**Reply by will9512** [reply]
This works for now 

Current solution (based on [ixtrader](https://github.com/ixtrader) code :

https://pastebin.com/79C7iX1T

dont forget to change the entries:
server_url = "https://immich.yurserver.xy/"
email = "[test@test.xy](mailto:test@test.xy)"
password = "yuorpassword"
person_id = ""
album_id = ""

**Reply by derpderpington** [reply]
Hey sorry I might be alittle slow, but I am trying to run this on Windows 11 with python and I keep getting an indentation error for right after this section you change all your variables. I have copied and pasted the code and I have tried downloading the code and running it as well but still this error persists. The indentation error is on line 92 with the login_response. Did you get this error or happen to know how to fix it? I would really like to get this working for shared face albums!

**Reply by will9512** [reply]
make sure " def main():  " is all the way to the left, then make sure every line after that up to and including  "login_response" line are 1 tab (or 4 spaces) from the far left. also, just copy paste the full script into chatgpt. it will fix it probably.

**Reply by derpderpington** [reply]
Thanks, ChatGPT was able to fix it, and it works!

**Reply by randyesperben** [reply]
Does this code work with the latest version of Immich and its API changes? also this https://github.com/romainrbr/immich-face-to-album/ ?

**Reply by romainrbr** [reply]
@randyesperben : https://github.com/romainrbr/immich-face-to-album/ works indeed with the latest version

**Reply by randyesperben** [reply]
> @randyesperben : https://github.com/romainrbr/immich-face-to-album/ works indeed with the latest version

Very thanks!!

**Reply by ptfuzi** [reply]
this is the feature i need to ditch everything else i'm using.

i see the command but how/where do install it?
i might need to find the API commands and all that but i just don't know where to put the commands.

**Comment by nick-holmquist** [comment]
Just searching to see if this was a feature and led me here. Commenting to add support for adding the feature.
I want to be able to share with a public album (or private) using a single face for an autogenerated album, or multiple faces. Would allow me to create an album that included specific family members or friends. Extra credit for a selection to allow "Only add to album if ALL faces are present in the photo" for example.

**Comment by TheWildBear** [comment]
Does someone know why this feature request is closed? And could someone please reopen it, since it would be a very nice feature for me and family and friends!

**Reply by bo0tzz** [reply]
It's closed because it's a duplicate of #1673, as mentioned upthread.

**Reply by SansGuidon** [reply]
the mentioned thread is locked down. Is there a link for following up on the upcoming related work?
thanks!

**Reply by bo0tzz** [reply]
If there are any updates they'll be in the thread.

**Comment by SansGuidon** [comment]
I did implement something like this through script in a matter of minutes thanks to the examples above and LLMs.
Could that be an intermediate solution while waiting for Immich to support this out of the box?

**Reply by TheWildBear** [reply]
That is a intermediate solution for me. Which I will use in the meantime. Just wanted to point out that the feature request is closed...

**Reply by SansGuidon** [reply]
that's indeed a pity, but maybe it's worth making a new issue/opening a new discussion for this, with a fresh crowd or voters.

**Comment by TrooP81** [comment]
just wanted to post that im also looking for this feature... hence i used vibe-coding to create this: https://github.com/TrooP81/Immich-ppl2album/ It works fine for me atleast =)

**Comment by ajb3932** [comment]
I actually made a docker container which will run every 60 mins (can be changed) and add faces to albums. It works across different accounts too which can share to an album. i.e. both parents photos of their child in one shared album.
Also includes an Unraid template.
[ajb3932/immich-partner-sharing](https://github.com/ajb3932/immich-partner-sharing)

**Reply by SansGuidon** [reply]
Thanks for sharing! Note that the correct link is https://github.com/ajb3932/immich-partner-sharing could you please update your original post? :)

**Reply by ajb3932** [reply]
Good spot, thanks! 👍

**Reply by ptfuzi** [reply]
That’s nice, I’m already using the face-to-album, wich was a pain to configure but now it’s working, but this looks much easier

**Reply by alangrainger** [reply]
Since the above image is no longer working after the Immich update, I've created a new one in Typescript which does the same thing. Credit to @ajb3932 for the idea:

https://github.com/alangrainger/immich-person-to-album

Probably the docs need improvement, but if you need something which will work right now, this will do the job.

**Comment by SansGuidon** [comment]
Are the scripts above still working with latest version of Immich? I have noticed my script stopped working properly and I have so many json parsing errors than I consider it could be caused by changes in Immich datamodel/api response. I'm curious about others.

**Reply by alangrainger** [reply]
I've created a new one in Typescript which does the same thing:

https://github.com/alangrainger/immich-person-to-album

Probably the docs need improvement, but if you need something which will work right now, this will do the job.

**Reply by SansGuidon** [reply]
Wow, thanks! I could adapt my existing python script based on your work, using API Key, and with help of Claude 4 model. And not the sync works again after months of issues

**Reply by projek01** [reply]
I just got this up and running today on my Synology. Thank you @alangrainger

**Comment by B85S5DSG** [comment]
I'm also looking for this feature.  But I'd also like to attach a user account to a face.  If I add a picture that matches a user, I'd like it automatically shared with them.

**Reply by ptfuzi** [reply]
There’s a python script for it, “immich face to album”

**Reply by alangrainger** [reply]
And a docker container: 

https://github.com/alangrainger/immich-person-to-album

**Reply by B85S5DSG** [reply]
The docker isn't working for me.  I have the API key, the correct IP, and it just times out.  I guess upon further inspection it just plain fails to pull.  Eventually it pulled, but I get the same timeout error.

**Reply by alangrainger** [reply]
@B85S5DSG make sure you can access your Immich API from inside that container. Probably a networking issue. Try this command from outside the container, making sure to replace that address with your Immich server address:

```bash
docker exec immich-person-to-album sh -c "wget -qO- http://192.168.0.20:2283/api/server/ping"
```

It will return `{"res":"pong"}` if Immich is accessible to the person-to-album container.

**Reply by B85S5DSG** [reply]
@alangrainger, thanks for the reply, but I decided to use the Python module method instead.  I just added that to crontab and it runs a script with all the names and folders that I want every 15 minutes.
